src/m365/spo/commands/field/field-list.ts

```diff
@@ -0,0 +1,103 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { urlUtil, validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  listUrl?: string;
+}
+
+class SpoFieldListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FIELD_LIST;
+  }
+
+  public get description(): string {
+    return 'Retrieves list of columns for specified list or site';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Id', 'Title', 'Group', 'Hidden'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    let listUrl: string = '';
+
+    if (args.options.listId) {
+      listUrl = `lists(guid'${encodeURIComponent(args.options.listId)}')/`;
+    }
+    else if (args.options.listTitle) {
+      listUrl = `lists/getByTitle('${encodeURIComponent(args.options.listTitle)}')/`;
+    }
+    else if (args.options.listUrl) {
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.listUrl);
+
+      listUrl = `GetList('${encodeURIComponent(listServerRelativeUrl)}')/`;
+    }
+
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/web/${listUrl}fields`,
+      headers: {
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get(requestOptions)
```

Comment:
Let's define the return type on `get` as `{ value: any[] }` so that we can design-time type checking on `value` and can refactor it in the future if needed.